R/functions.R

```diff
@@ -5,22 +5,34 @@
 #' @export
 make_entity_name <- function(df){
   first <- df[1,]
-  name <- glue::glue("{first$lastname} {first$journal} {first$year} (Pubmed ID {first$pmid})")
+  # Entity names must be < 256 characters; shorten author, journal section
+  # Need to leave space for year and pubmed ID
+  # Arbitrarily set to 200 characters
+  short_name <- stringr::str_trunc(
+    glue::glue("{first$lastname} {first$journal}"),
+    width = 200
+  )
+  name <- glue::glue("{short_name} {first$year} (Pubmed ID {first$pmid})")
```

Comment:
That could be added. The main reason why I didn't is that years are always 4 characters (at least for the relevant future) and the Pubmed ID is generally 8 characters. These two things added with the extra formatting take up 26 characters. Since I cut down the first bit to 200 characters, that leaves about 36 characters of buffer, in case Pubmed increases the length of their ids.

If you would like an extra check here, what would you like it to look like? Another truncation to 256 or throwing an error (i.e. stopping that one from finishing)?

Reply:
You make a great point! No need to add that check.